src/controllers/remove.director.controller.ts

```diff
@@ -27,9 +37,47 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
   try {
     //Hardcoded for now
     req.params[urlParams.PARAM_SUBMISSION_ID] = "645d1188c794645afe15f5cc";
+
+    const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
+    const submissionId = "645d1188c794645afe15f5cc";
+    const session: Session = req.session as Session;
+
+    const validationStatus: ValidationStatusResponse = await getValidationStatus(session, transactionId, submissionId);
+    const errorMessage = retrieveErrorMessageToDisplay(validationStatus);
+
+
+    if (errorMessage) {
+      const error = formatValidationError(errorMessage);
+
+      // Bypass the direct use of variables with dashes that
+      // govukDateInput adds for day, month and year field
+      
+      const dates = {
+        [RemovalDateKey]: RemovalDateKeys.reduce((o, key) => Object.assign(o, { [key]: req.body[key] }), {})
+      };
+
+      const backLink = OFFICER_FILING + req.route.path.replace(REMOVE_DIRECTOR_PATH_END, ACTIVE_OFFICERS_PATH_END);
+
+      return res.render(Templates.REMOVE_DIRECTOR, {
+        backLinkUrl: backLink,
+        templateName: Templates.REMOVE_DIRECTOR,
+        ...req.body,
+        ...dates,
+        error
+      });
+
+    }
+
     const nextPageUrl = urlUtils.getUrlToPath(REMOVE_DIRECTOR_CHECK_ANSWERS_PATH, req);
     return res.redirect(nextPageUrl);
   } catch (e) {
     return next(e);
   }
 };
+
+export function formatValidationError(errorMessage: string): FormattedValidationErrors {
+  const errors = { errorMessage: [] } as any;
```

Comment:
Maybe Record<string, any>

Reply:
The data structure has to be consistent with existing error validation so an array must be used here

Reply:
However, at the bottom we add a new key which accepts value string. Hence, transforming the the datatype Record<string, Array<string>> to Record<string, any>. Without that errors = 9 will also be acceptable.